File: Backend_public/app/Models/Cliente.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Cliente extends Model
{
    use HasFactory;

    protected $table = 'CLIENTES';
    protected $primaryKey = 'codigo';
    protected $fillable =['codigo','nombreUsuario','nombre','apellido1','apellido2','telefono','email','codigoPostal','password'];
    public $timestamps = false;


}

File: Backend_private/src/main/java/coworking/backend_private/Entidad/Cliente.java
package coworking.backend_private.Entidad;

import lombok.Data;
import org.mindrot.jbcrypt.BCrypt;

import javax.persistence.*;
import java.io.Serializable;


@Data
@Entity
@Table(name="CLIENTES")
public class Cliente implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer codigo;
    private String nombreUsuario;
    private String password;
    private String nombre;
    private String apellido1;
    private String apellido2;
    private String telefono;
    private String email;
    private String codigoPostal;
    private String rol = "cliente";
    private boolean alta = true;

    private String hashPassword(String passwordNotEncrypted) {
        return BCrypt.hashpw(passwordNotEncrypted, BCrypt.gensalt());
    }

    public void setPassword(String password) {
        this.password = hashPassword(password);
    }
}
